parsec/core/fs/workspacefs/workspacefs.py

```diff
@@ -423,14 +408,14 @@ async def copyfile(
         Raises:
             FSError
         """
-        await self.touch(target_path, exist_ok=exist_ok)
-        offset = 0
-        while True:
-            buff = await self.read_bytes(source_path, length, offset * length)
-            if not buff:
-                break
-            await self.write_bytes(target_path, buff, offset * length)
-            offset += 1
+        write_mode = "wb" if exist_ok else "xb"
```

Comment:
Can use "wb" mode in this case because it will also create the file if it doesn't exist